rules/aip0203/field_behavior_required.go

```diff
@@ -23,47 +23,62 @@ import (
 	"github.com/jhump/protoreflect/desc"
 )
 
-var minimumRequiredFieldBehavior = stringset.New(
+var fbs = stringset.New(
 	"OPTIONAL", "REQUIRED", "OUTPUT_ONLY", "IMMUTABLE",
 )
 
 var fieldBehaviorRequired = &lint.MethodRule{
 	Name: lint.NewRuleName(203, "field-behavior-required"),
 	LintMethod: func(m *desc.MethodDescriptor) []lint.Problem {
-		// we only check requests, as OutputTypes are always
-		// OUTPUT_ONLY
-		it := m.GetInputType()
-		return checkFields(it)
+		req := m.GetInputType()
+		ps := problems(req)
+		if len(ps) == 0 {
+			return nil
+		}
+
+		return ps
 	},
 }
 
-func checkFields(m *desc.MessageDescriptor) []lint.Problem {
-	problems := []lint.Problem{}
+func problems(m *desc.MessageDescriptor) []lint.Problem {
+	var ps []lint.Problem
+
 	for _, f := range m.GetFields() {
-		asMessage := f.GetMessageType()
-		if asMessage != nil {
-			problems = append(problems, checkFields(asMessage)...)
+		if r := utils.GetResource(m); r != nil && f.GetName() == "name" {
+			continue
+		}
+
+		p := checkFieldBehavior(f)
+		if p != nil {
+			ps = append(ps, *p)
+		}
+
+		if mt := f.GetMessageType(); mt != nil {
+			ps = append(ps, problems(mt)...)
 		}
-		problems = append(problems, checkFieldBehavior(f)...)
 	}
-	return problems
+
+	return ps
 }
 
-func checkFieldBehavior(f *desc.FieldDescriptor) []lint.Problem {
```

Comment:
nit: this was an intentional choice to ensure that one could easily attach new errors, without the refactor to a list.

But i'm not strongly opinionated if you change it.